src/js/dataobjects.js

```diff
@@ -21,6 +21,37 @@ export class Geography {
     }
 }
 
+export class IndicatorHelper {
+    static fixMetadata(indicator) {
+        if (indicator.metadata == undefined)
+            indicator.metadata = {
+                source: "",
+                description: "",
+                licence: {
+                    name: "",
+                    url: ""
+                }
+
+            }
+
+        return indicator
+    }
+
+    static addDefaultConfiguration(indicator) {
+        if (indicator.chartConfiguration == undefined)
+            indicator.chartConfiguration = {}
+
+        indicator.chartConfiguration = fillMissingKeys(defaultValues.chartConfiguration, indicator.chartConfiguration)
```

Comment:
I don't think I can just return the configuration - I still need to set it to indicator. Since fillMissingKeys is idempotent. It doesn't modify the original config object

Reply:
just read through all of your coments - it makes sense, and is quite elegant, thanks